multi-layer-app-config/README.md

```diff
@@ -0,0 +1,193 @@
+```yaml
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+spec:
+  catalog: giantswarm
+  config:
+    configMap:
+      name: ingress-controller-values
+      namespace: m2m01
+  configs: []  # <-- new
+  name: nginx-ingress-controller-app
+  namespace: kube-system
+  userConfig:
+    configMap:
+      name: nginx-ingress-controller-app-user-values
+      namespace: m2m01
+  version: 2.7.0
+```
+
+Using the new list instead of `config` and `userConfig`, works exactly the same as the one above:
+
+```yaml
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+spec:
+  catalog: giantswarm
+  configs:  # <-- new
+    - kind: configMap
+      name: ingress-controller-values
+      namespace: m2m01
+    - kind: configMap
+      name: nginx-ingress-controller-app-user-values
+      namespace: m2m01
+  name: nginx-ingress-controller-app
+  namespace: kube-system
+  version: 2.7.0
+```
+
+Using the new list only and referencing more than 2 objects:
+
+```yaml
+apiVersion: application.giantswarm.io/v1alpha1
+kind: App
+spec:
+  catalog: giantswarm
+  configs:  # <-- new
+    - kind: configMap
+      name: ingress-controller-values
+      namespace: m2m01
+    - kind: configMap
+      name: nginx-ingress-controller-app-user-values
+      namespace: m2m01
+    - kind: secret
+      name: nginx-ingress-controller-admin-login
+      namespace: m2m01
+    - kind: configMap
+      name: nginx-ingress-controller-admin-account
+      namespace: m2m01
+  name: nginx-ingress-controller-app
+  namespace: kube-system
+  version: 2.7.0
+```
+
+#### Pros
+
+- solves the problem everywhere, in App CR and in GitOps
+- backward compatible
+- easy to implement
+- makes `app-operator` more like `helm-controller` from Flux, which may make replacing `chart-operator` with it easier
+
+#### Cons
+
+- we're solving a problem already solved in Flux's `HelmRelease`
+- we have to implement it
+
+### Implementing ConfigMap key merging in `kustomize`
+
+Currently, `kustomize` can't merge keys within a `ConfigMap`. Still, this is a 'wanted' feature. We might just
+implement this in `kustomize`.
+
+#### Pros
+
+- solves the problem for GitOps
+- fame and glory in the community for implementing a needed feature
+- getting to know `kustomize`'s code base
+
+#### Cons
+
+- we have to implement it on an unknown code base - might be a bigger challenge
+- this solves the problem for GitOps only (well, any tool using `kustomize`)
+
+### Drop App CR and switch to HelmRelease
+
+This seems unrealistic, as we really a lot on app platform and its features. Still, bypassing App CR in GitOps
+scenario would solve the problem.
+
+#### Pros
+
+- we don't implement anything in code
+
+#### Cons
+
+- this might mean phasing out app platform operators or living in two worlds (with and without app platform)
+- App CRs and app-platform are deeply integrated into our product, we can't get rid of it easily
+
+### Dropping the 'values' key in AppCR's CMs/Secrets and moving them to top level
```

Comment:
I am not very fond of this one. Leaning more towards App CR or getting rid of app platform if that makes sense.